cmd/generate.go

```diff
@@ -201,9 +206,13 @@ func findPreviousRelease(r *git.Repository, v *semver.Version) (*semver.Version,
 	if maxRef == nil {
 		return nil, plumbing.ZeroHash, errors.New("no previous release found")
 	}
-	tag, err := r.TagObject(maxRef.Hash())
+	tag, err := r.TagObject(maxRef.Hash()) // only returns Annotated Tags
 	if err != nil {
-		return nil, plumbing.ZeroHash, errors.Wrap(err, "could not get tag object from maxRef")
+		if !lightweightOK {
+			return nil, plumbing.ZeroHash, errors.Wrap(err, "could not get tag object from maxRef")
+		} else {
+			return maxVersion, maxRef.Hash(), nil
```

Comment:
Does `Tag` work here? https://github.com/src-d/go-git/blob/8b0c2116cea2bbcc8d0075e762b887200a1898e1/repository_test.go#L2173

Reply:
Can you elaborate a bit more? I'm not quite sure what you're asking.